### ISSUE TITLE
merge_toas cannot merge TOAs objects when one has pulse numbers and the other does not

### ISSUE THREAD

**Issue by aarchiba** [nanograv/PINT#1034]
```
E           TypeError: merge_TOAs() cannot merge. Inconsistent numbers of table columns: [9, 10]
```

**Comment by scottransom** [comment]
@aarchiba Do you think this is a bug or just a simple statement of fact?  I can see both sides given that a user can add any columns they want at any time.  And a pulse number column is just another column (albeit one that can easily be re-calculated if needs be).

**Comment by aarchiba** [comment]
Well, concretely, you can try to load in and merge two different `tim` files and have it fail because one has a pulse number column and the other does not. (Which is how I found this.) And since pulse number columns already permit NaNs to indicate that some pulse numbers are not known, there is a fairly easy fix for this. Even if we were to decide that this shouldn't be possible, a more sensible error message would be a help.

There certainly is more work to be done around partial sets of pulse numbers - currently we can't use pulse numbers unless all TOAs have them, but that is in principle both possible and useful to fix (for example, lock in the known and trusted part of a timing solution then add some new bits whose pulse numbers are unknown. Unfortunately the way we handle pulse numbers and mean subtraction right now makes this nontrivial to sort out.

**Comment by abhisrkckl** [comment]
Solved by #1398
